marklogic-client-api/src/main/java/com/marklogic/client/impl/RowManagerImpl.java

```diff
@@ -377,7 +378,13 @@ private RESTServiceResultIterator makeRequest(
     addDatatypeStyleParam(params,     datatypeStyle);
     addRowStructureStyleParam(params, rowStructureStyle);
 
-// QUESTION: outputMimetypes a noop?
+    if (requestPlan instanceof PlanBuilderBase.AccessPlanBase) {
+      Map<String, AbstractWriteHandle> contentParams = ((PlanBuilderBase.AccessPlanBase)requestPlan).getContentParams();
```

Comment:
This seems like the right place to insert the multipart branch - i.e. if the plan doesn't have any content params, then we can do a regular post as there's no need for multipart. But if there's at least one content param, we know we need a multipart request, so we go through a different branch of code.